application/helpers/api_helper.php

```diff
@@ -326,6 +326,11 @@ function filter_remote_url($url, $allowed_schemes = array('http', 'https')) {
         {
```

Comment:
not related to this issue, but why do we block all IPs? should we block only internal IPs?

Reply:
🤷‍♂️ This is what you get when code is written without tests that act as documentation... The reasons are lost to time!

Reply:
This code is probably used in more places than it should. Assuming this is _only_ for the validator tool, allowing IP addresses makes this very complex. The fewer options we give users the better in order to reduce the security scope and surface area.

I think its fair to say that if the dashboard is crawling a DCAT-US JSON file, that file should be hosted on a URL with a domain and TLS on default ports. As a fallback, users can always upload a file manually for validation if they can't meet those requirements.

Reply:
I dont see why allowing IP makes it less secure. It is the flag `FILTER_FLAG_NO_PRIV_RANGE | FILTER_FLAG_NO_RES_RANGE` that ultimately keep it secure. Any IP not in that range can be registered by hacker and tested on the tool. I agree it creates more work for hackers.

It is the flaws in the IP check makes it less secure, especially combined with how the if/else condition check and how we let through unresolved DNS. Entries like **0x7f000001**:80 or **0**:80 pass the flawed IP check, then pass as unresolved, that is how they become issues.

That is why I am questioning the use IP check and later on letting unresolved entry pass the check.